### PR TITLE
Fix hive tests peering issue

### PR DESCRIPTION
**Pull request by smartprogrammer93**
## Changes

- Ignore impossible block numbers and timestamps

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or a feature that would cause existing functionality not to work as expected)
- [ ] Documentation update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional or API changes)
- [ ] Build-related changes
- [ ] Other: _description_ 

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Notes on testing

Tested with devp2p hive tests and passing

https://github.com/NethermindEth/nethermind/actions/runs/3887919285/jobs/6634708147

### CODE REVIEW

**Comment by smartprogrammer93** [comment]
Unit tests added through adding shanghai to the list of transitions in mainnetSpecProvider (shanghai has impossible transition Timestamp)

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs [hunk 1]
```diff
@@ -205,15 +205,15 @@ private static void LoadTransitions(ChainSpecJson chainSpecJson, ChainSpec chain
             chainSpec.MuirGlacierNumber = chainSpec.Ethash?.DifficultyBombDelays.Count > 2 ?
                 chainSpec.Ethash?.DifficultyBombDelays.Keys.ToArray()[2]
                 : null;
-            chainSpec.BerlinBlockNumber = chainSpec.Parameters.Eip2929Transition ?? (long.MaxValue - 1);
-            chainSpec.LondonBlockNumber = chainSpec.Parameters.Eip1559Transition ?? (long.MaxValue - 1);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:209]: Why this was removed?

File: src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs [hunk 2]
```diff
@@ -205,15 +205,15 @@ private static void LoadTransitions(ChainSpecJson chainSpecJson, ChainSpec chain
...
+            chainSpec.BerlinBlockNumber = chainSpec.Parameters.Eip2929Transition;
+            chainSpec.LondonBlockNumber = chainSpec.Parameters.Eip1559Transition;
             chainSpec.ArrowGlacierBlockNumber = chainSpec.Ethash?.DifficultyBombDelays.Count > 4 ?
                 chainSpec.Ethash?.DifficultyBombDelays.Keys.ToArray()[4]
                 : null;
             chainSpec.GrayGlacierBlockNumber = chainSpec.Ethash?.DifficultyBombDelays.Count > 5 ?
                 chainSpec.Ethash?.DifficultyBombDelays.Keys.ToArray()[5]
                 : null;
-            chainSpec.ShanghaiTimestamp = chainSpec.Parameters.Eip3651TransitionTimestamp ?? (long.MaxValue - 1);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:216]: Why not null for consistency?

Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:216]: I guess it should be

**Review by marcindsobczak — APPROVED** [review]

**Pull request merged**